v2/cmd/csv.go

```diff
@@ -0,0 +1,280 @@
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	// csvCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// csvCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+func csvImp() (err error) {
+	config, err := configmodule.Load("")
+	if err != nil {
+		return err
+	}
+
+	if config.Module.LicenseDB.Path == "" {
+		config.Module.LicenseDB.Path, err = defaultLicenseDB()
+		if err != nil {
+			klog.Exit(fmt.Errorf("licenseDB.path is empty, also failed to get defaulut licenseDB path: %w", err))
+		}
+		klog.V(2).InfoS("Config: use default license DB")
+	}
+	klog.V(2).InfoS("Config: license DB path", "path", config.Module.LicenseDB.Path)
+
+	goModuleRefs, err := deps.ListModulesInGoBinary(config.Module.Go.Binary.Path)
+	if err != nil {
+		return err
+	}
+	goModules, err := deps.JoinModuleRefWithLocalModules(goModuleRefs)
+	if err != nil {
+		return err
+	}
+	mainModuleAbsPath, err := filepath.Abs(config.Module.Go.Path)
+	if err != nil {
+		return err
+	}
+	mainModule := []deps.GoModule{{
+		ImportPath: config.Module.Go.Module,
+		SrcDir:     mainModuleAbsPath,
+	}}
+	mainModule = append(mainModule)
+	goModules = append(mainModule, goModules...)
+	klog.InfoS("Done: found dependencies", "count", len(goModules))
+	if klog.V(3).Enabled() {
+		for _, goModule := range goModules {
+			klog.InfoS("dependency", "module", goModule.ImportPath, "version", goModule.Version, "srcDir", goModule.SrcDir)
+		}
+	}
+	f, err := os.Create(config.Module.Csv.Path)
+	if err != nil {
+		return errors.Wrapf(err, "Creating license csv file")
+	}
+	defer func() {
+		closeErr := f.Close()
+		if err == nil {
+			// When there are no other errors, surface close file error.
+			// Otherwise file content may not be flushed to disk successfully.
+			err = closeErr
+		}
+	}()
+	_, err = f.WriteString("# Generated by https://github.com/google/go-licenses/v2. DO NOT EDIT.\n")
+	if err != nil {
+		return err
+	}
+	licenseCount := 0
+	errorCount := 0
+	for _, goModule := range goModules {
+		report := func(err error, args ...interface{}) {
+			errorCount = errorCount + 1
+			errorArgs := []interface{}{"module", goModule.ImportPath}
+			errorArgs = append(errorArgs, args...)
+			klog.ErrorS(err, "Failed", errorArgs...)
+		}
+		var override configmodule.ModuleOverride
+		for _, o := range config.Module.Overrides {
+			if o.Name == goModule.ImportPath {
+				override = o
+			}
+		}
+		// When override.Version == "", the override apply to any version.
+		if override.Version != "" && override.Version != goModule.Version {
+			report(fmt.Errorf("override version mismatch: version %q!=%q", goModule.Version, override.Version))
+			continue
+		}
+		if override.Skip {
+			klog.InfoS("Skipped", "module", goModule.ImportPath)
+			continue
+		}
+		repo, errGetGithubRepo := goutils.GetGithubRepo(goModule.ImportPath)
```

Comment:
Reading the issue some more, looks like they aren't going forward with the meta tag approach. :(

Perhaps we can still leverage https://pkg.go.dev/golang.org/x/pkgsite@v0.0.0-20210618174102-9b1c9bd2b0c3/internal/source though?

Reply:
P0

Yep, pkgsite looks like the right fit!
How did you find it? If I knew it in advance, I would be able to avoid writing my own implementation. (I did search initially, but didn't find it)